src/adapter_covid19/corporate_bankruptcy.py

```diff
@@ -0,0 +1,220 @@
+        theta = sp.pi / self.beta
+        self.sinc_theta = sp.sin(theta) / theta
+        self.large_cap_cash_surplus_months = large_cap_cash_surplus_months
+        self.small_cap_cash_buffer: Mapping[Sector, float] = {}
+        self.large_cap_pct: Mapping[Sector, float] = {}
+        self.employee_compensation: Mapping[Sector, float] = {}
+        self.taxes_minus_subsidies: Mapping[Sector, float] = {}
+        self.capital_consumption: Mapping[Sector, float] = {}
+        self.value_added: Mapping[Sector, float] = {}
+        self.lcap_clipped_cash_buffer: Mapping[Sector, float] = {}
+        self.sme_clipped_cash_buffer: Mapping[Sector, float] = {}
+
+    def load(self, reader: Reader) -> None:
+        io_df = reader.load_csv("input_output").set_index("Sector")
+        self.small_cap_cash_buffer = SectorDataSource("smallcap_cash").load(reader)
+        self.large_cap_pct = SectorDataSource("largecap_pct_turnover").load(reader)
+        self.employee_compensation = {
+            Sector[k]: v for k, v in io_df.employee_compensation.to_dict().items()
+        }
+        self.taxes_minus_subsidies = {
+            Sector[k]: v for k, v in io_df.taxes_minus_subsidies.to_dict().items()
+        }
+        self.capital_consumption = {
+            Sector[k]: v for k, v in io_df.capital_consumption.to_dict().items()
+        }
+        outflows = (
+            io_df.employee_compensation
+            + io_df.taxes_minus_subsidies
+            + io_df.capital_consumption
+        )
+        gross_operating_surplus = (
+            io_df.net_operating_surplus + io_df.capital_consumption
+        )
+        lcap_cash_buffer = (
+            gross_operating_surplus
+            * np.array([self.large_cap_pct[s] for s in Sector])
+            * self.large_cap_cash_surplus_months
+            / 12
+        )
+        sme_factor = np.array(
+            [
+                (self.small_cap_cash_buffer[s] / self.sinc_theta)
+                / 365
+                * (1 - self.large_cap_pct[s])
+                for s in Sector
+            ]
+        )
+        sme_cash_buffer = outflows * sme_factor
+        lcap_clipped_cash_buffer = lcap_cash_buffer.apply(lambda x: max(x, 0))
+        sme_clipped_cash_buffer = sme_cash_buffer.apply(lambda x: max(x, 0))
+        self.lcap_clipped_cash_buffer = {
+            Sector[k]: v for k, v in lcap_clipped_cash_buffer.to_dict().items()
+        }
+        self.sme_clipped_cash_buffer = {
+            Sector[k]: v for k, v in sme_clipped_cash_buffer.to_dict().items()
+        }
+        value_added = (
+            io_df.net_operating_surplus
+            + io_df.employee_compensation
+            + io_df.taxes_minus_subsidies
+            + io_df.capital_consumption
+        )
+        self.value_added = {Sector[k]: v for k, v in value_added.to_dict().items()}
+
+    def _get_mean_cash_buffer_days(
+        self,
+        lcap: bool,
+        gdp_discount: Optional[Mapping[Sector, float]],
```

Comment:
> Optional[X] is equivalent to Union[X, None].

>Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional

(https://docs.python.org/3/library/typing.html#typing.Optional)